src/testing/cluster/storage_checker.zig

```diff
@@ -171,5 +178,42 @@ pub fn StorageCheckerType(comptime Storage: type) type {
             }
             return checksum;
         }
+
+        fn checksum_grid(checker: *Self, superblock: *const SuperBlock) u128 {
+            assert(superblock.working.vsr_state.sync_op_max == 0);
+
+            const free_set_zone = vsr.SuperBlockTrailer.free_set.zone();
+            const free_set_offset = vsr.Zone.superblock.offset(free_set_zone.start_for_copy(0));
+            const free_set_size = superblock.working.free_set_size;
+            const free_set_buffer = superblock.storage.memory[free_set_offset..][0..free_set_size];
+            assert(vsr.checksum(free_set_buffer) == superblock.working.free_set_checksum);
+
+            checker.free_set.decode(@alignCast(free_set_buffer));
+            defer checker.free_set.reset();
+
+            var checksum: u128 = 0;
+            var blocks_missing: usize = 0;
+            var blocks_acquired = checker.free_set.blocks.iterator(.{});
+            while (blocks_acquired.next()) |block_address_index| {
+                const block_address: u64 = block_address_index + 1;
+                const block = superblock.storage.grid_block(block_address) orelse {
+                    log.err("{}: checksum_grid: missing block_address={}", .{
+                        superblock.replica_index.?,
+                        block_address,
+                    });
+
+                    blocks_missing += 1;
+                    continue;
+                };
+
+                const block_header = schema.header_from_block(block);
+                assert(block_header.op == block_address);
+
+                checksum ^= vsr.checksum(block[0..block_header.size]);
```

Comment:
I may be paranoid, but let's add some kind of a counter or nonce here, so that identical blocks don't cancel each other out.